astropy/extern/ply/lex.py

```diff
@@ -179,7 +179,7 @@ def writetab(self, lextab, outputdir=''):
         with open(filename, 'w') as tf:
             tf.write('# %s.py. This file automatically created by PLY (version %s). Don\'t edit!\n' % (basetabmodule, __version__))
             tf.write('_tabversion   = %s\n' % repr(__tabversion__))
-            tf.write('_lextokens    = set(%s)\n' % repr(tuple(sorted(self.lextokens))))
+            tf.write('_lextokens    = {%s}\n' % repr(tuple(sorted(self.lextokens)))[1:-1])
```

Comment:
This is a bundled external package - I think this would need a PR upstream!

Reply:
Yeah please don't touch anything in `extern`.

Reply:
It looks like all we need to do is upgrade ply to the 2020 version after it went through a major refactor.

Reply:
Sounds good - that's then separate from this PR. So, I would suggest not touching the lex/parse tables here, and excluding them in the pre-commit PR.

Reply:
Yikes. So that doesn't work. The 2020+ version of "PLY no longer writes table files or cached data.  If you want this,
it's your responsibility to serialize the parser tables.  Use pickle." This is a major breaking change.

Reply:
https://github.com/dabeaz/ply/blob/master/CHANGES

Reply:
I found an example, https://github.com/asottile/pyupgrade/issues/622#issuecomment-1088393650

Reply:
@nstarman - it may well be OK, and of course we do not actually have to update at all. For now, maybe just raise a separate issue?